### PR TITLE
Move `seed-bootstrap` chart and monitoring-related logic in `Seed` controller into `monitoring` component

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area dev-productivity
/kind cleanup

**What this PR does / why we need it**:
Move `seed-bootstrap` chart and monitoring-related logic in `Seed` controller into `monitoring` component for better dev-productivity (incl. usage of go embed).

In https://github.com/gardener/gardener/pull/8243, I'll also move the `seed-monitoring` chart.

Part of https://github.com/gardener/gardener/issues/6016

**Special notes for your reviewer**:
This PR explicitly just moves code and does not introduce tests or usage of `ManagedResource` since all this should be replaced in the scope of GEP-19.

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: pkg/component/monitoring/bootstrap.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gardener/gardener/pkg/component"
+)
+
+// Values is a set of configuration values for the monitoring components.
+type Values struct {
+}
+
+// New creates a new instance of DeployWaiter for the monitoring components.
```
Comment (acumino) [inline pkg/component/monitoring/bootstrap.go]: Only deployer is created.

```suggestion
// New creates a new instance of Deployer for the monitoring components.
```

File: pkg/operation/botanist/botanist.go [hunk 2]
```diff
@@ -41,8 +41,6 @@ const DefaultInterval = 5 * time.Second
 var (
 	// ChartsPath is an alias for charts.Path. Exposed for testing.
 	ChartsPath = charts.Path
-
-	ingressTLSCertificateValidity = 730 * 24 * time.Hour // ~2 years, see https://support.apple.com/en-us/HT210176
```
Comment (acumino) [inline pkg/operation/botanist/botanist.go:45]: This value is being used in other places also, can you harmonize everywhere to use newly defined constant?

File: pkg/component/monitoring/bootstrap.go [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+		},
+		"ingress": map[string]interface{}{
+			"authSecretName": b.values.GlobalMonitoringSecret.Name,
+		},
+	})
+
+	return b.chartApplier.ApplyFromEmbeddedFS(ctx, chart, chartPath, b.namespace, "monitoring", values, applierOptions)
+}
+
+func (b *bootstrapper) Destroy(ctx context.Context) error {
```
Comment (acumino) [inline pkg/component/monitoring/bootstrap.go:274]: Is the deletion of resources on seed deletion not required?

File: pkg/operation/botanist/botanist.go (hunk 2 above)
Comment (rfranzke) [reply pkg/operation/botanist/botanist.go:45]: This is done in the follow-up PR (https://github.com/gardener/gardener/pull/8243)

File: pkg/component/monitoring/bootstrap.go (hunk 3 above)
Comment (rfranzke) [reply pkg/component/monitoring/bootstrap.go:274]: Technically it is, but it wasn't implemented before, and this PR only focuses on moving the chart into the component package. It does not add new logic/behaviour. This cleanup topic should probably be followed-up separately.

**Review by acumino — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**